src/extension.ts

```diff
@@ -1,15 +1,106 @@
-import { IGatherProvider } from "./types";
+"use strict";
+import * as vscode from "vscode";
+import { Constants, IGatherProvider, KernelState, KernelStateEventArgs, Telemetry } from "./types/types";
 import { GatherProvider } from "./gather";
-import { ExtensionContext } from "vscode";
+import { IJupyterExtensionApi } from "./types/jupyter";
+import { sendTelemetryEvent } from "./telemetry";
+import * as localize from './localize';
 
-export function activate(context: ExtensionContext) {
-  let api = {
-    getGatherProvider(): IGatherProvider {
-      return new GatherProvider(context.extensionPath);
-    },
-  };
+const registeredButtons: vscode.Disposable[] = [];
 
-  return api;
+export async function activate() {
```

Comment:
Just as a bit of general feedback. This function is a bit long and multipurpose. It might read better with a comment or two for each chunk of code. Or maybe breaking out a few things like the button registration into a sub-function. Not sure if I'd usually expect to see the implementation of button actions in an extensions base activate function.